src/site/content/en/blog/webxr-pt1/index.md

```diff
@@ -0,0 +1,317 @@
+an `XRSession` instance. In early versions of the API, a script had to request a
+device before requesting a session. Now, the device is acquired implicitly.
+
+### Entering a session
+
+After getting a session, I need to start it and enter it. But first, I need to
+set up a few things. A session needs an `onend` event handler so that the app or
+web page can be reset when the user exits.
+
+I'll also need a `<canvas>` element to draw my scene on. It needs to be an
+XR-compatible WebGL context. All drawing is done using the WebGL API or a
+WebGL-based framework such as [Three.js](https://threejs.org/).
+
+Now that I have a place to draw, I need a source of content to draw on
+it. For that, I create an instance of `XRWebGLLayer`. I associate it with the
+canvas by calling `XRSession.updateRenderState()`.
+
+Once I'm in a session, I need a way of determining where things are in virtual
+reality. I'll need a reference space. A `'local-floor'` reference space is one
+where the origin is located near the viewer and the y-axis is 0 at floor level
+and is not expected to move. There are [other types of reference
+spaces](https://developer.mozilla.org/en-US/docs/Web/API/XRSession/requestReferenceSpace),
+but that is a more complicated topic than I can go into here. I save the
+reference space to a variable because I'll need it when I draw to the screen.
+
+```js
+function onSessionStarted(xrSession) {
+  xrSession.addEVentListener('end', onSessionEnded);
+
+  let canvas = document.createElement('canvas');
+  gl = canvas.getContext('webgl', { xrCompatible: true });
+
+  xrSession.updateRenderState({
+    baseLayer: new XRWebGLLayer(session, gl)
+  });
+
+  xrSession.requestReferenceSpace('local-floor')
+  .then((refSpace) => {
+    xrRefSpace = refSpace;
+    xrSession.requestAnimationFrame(onXRFrame);
+  });
+}
+```
+
+After getting a reference space, I call `XRSession.requestAnimationFrame()`.
+This is the start of presenting virtual content, which is done in the frame
+loop.
+
+### Running a frame loop
+
+The frame loop is a user-agent controlled infinite loop in which content is
+repeatedly drawn to the screen. Content is drawn in discrete blocks called
+frames. The succession of frames creates the illusion of movement. The number of
+frames per second currently varies between devices, and may be anything from 60
+to 144, but that has no bearing on your code.
+
+The basic process for the frame loop is:
+
+<ol>
+  <li>Call <code>XRSession.requestAnimationFrame()</code>. In response, the user agent invokes the <code>XRFrameRequestCallback</code>, which is defined by you.</li>
+  <li>Inside your callback function:
+    <ol>
+      <li>Call <code>XRSession.requestAnimationFrame()</code> again.</li>
+      <li>Query for the position (called a pose in WebXR) of the viewer.</li>
+      <li>Draw content from the viewer's point of view.</li>
+      <li>Process user input.</li>
+    </ol>
+  </li>
+</ol>
```

Comment:
I think Markdown can handle a list like this. If this doesn't work:

```
1. A
1. B
   1. C
   1. D
1. E
```

Then I'm pretty sure this will work:

```
1. A
1. B
     1. C
     1. D
1. E
```

(Only difference being that C and D are indented 2 more spaces)

Reply:
I tried it. It didn't work.

Reply:
Good to know. Thanks.